In [1]:
pip install statsmodels

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: C:\Users\user\AppData\Local\Programs\Python\Python311\python.exe -m pip install --upgrade pip


In [2]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: C:\Users\user\AppData\Local\Programs\Python\Python311\python.exe -m pip install --upgrade pip


In [1]:
import numpy as np

## TASK 1

<b>Q1.</b>Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05.

##### ANSWER:

<b>Ho</b> = μ ≤ 11 minutes  (NULL Hypothesis)

<b>Ha</b> = μ > 11 minutes (Alternative Hypothesis)

<b>x̄</b> = 11.5 minutes

<b>n</b> = 1000

<b>σ</b> = 2.3 minutes

<b>α</b> = 0.05

<u><b>Statistical Test</b></u>
Since we have a sample of 1000, which is greater than 30, and we have a sample mean, standard deviation, and want to compare the sample mean to a population value ,so we can use Z-test.

<table style="border:true">
<td><b>Formula Z=(x̄ - μ)/(σ/√n)</b></td>
</table>

Where: 
<ul>
<li>x̄ = sample mean (11.5 minutes)</li>
<li>μ = population mean (11 minutes)</li>
<li>σ = population standard deviation (2.3 minutes)</li>
<li>n = sample size (1000).</li>
</ul>

In [3]:
import math

In [4]:
Z = (11.5-11)/(2.3/math.sqrt(1000))
Z

6.874516652539955


### Find Critical Value:
Since we're conducting a one-tailed test (since we're testing if the average time is more than 11 minutes), we will look up the critical value for a z-score corresponding to an alpha level of 0.05 from z-distribution table.
<p>corresponds to an alpha level of 0.05</p>
<p>Critical value=1.645</p>

## Conclusion:
Based on the test, we do not have enough evidence to reject the null hypothesis. Therefore, we conclude that the data do not contradict the psychologist's claim that the average time working mothers spend talking to their children is at least 11 minutes per day at the 0.05 significance level.

# TASK 2

<b>Q2.</b> A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

<B>ANSWER:-</B>

<b>
H</b>o = μ ≥ 5 minute (Null Hypothesis)sv  in-1) = 39

<b>Ha</b> = μ < 5 minutes (Alternative Hypothesis)

<b>x̄</b> = 4.6 minutes

<b>n</b> = 40

In [11]:
population_tmean = 5

In [12]:
sample_tmean =4.6

In [13]:
sample_size = 40

In [14]:
std_t = 0.8

<b>σ</b> = 0.8 minutes

In [15]:
alpha = 0.05

<u><b>Statistical Test</b></u>
Here the population std is unknown, so we can use t-test for this.
If the sample size is small (typically <30) and the population standard deviation is unknown, the t-test is appropriate. However, for larger sample sizes (usually greater than 30), the t-test can also be used even if the population standard deviation is unknown due to the Central Limit Theorem, which states that the distribution of sample means approaches a normal distribution as the sample size increases.

<table style="border:true">
<td><b>Formula t_test_statistic=(x̄ - μ) / (σ / √n)</b></td>
</table>


Where: 
<ul>
<li>x̄ = sample mean (4.6 minutes)</li>
<li>μ = population mean (5 minutes)</li>
<li>σ = Sample standard deviation (0.8 minutes)</li>
<li>n = sample size (40).</li>
    <li>α = Significance level (0.05(5%)).</li>
<li>n-1 = Degree of freedom(39).</li>
</ul>

In [16]:
t_test =  (sample_tmean - population_tmean) / (0.8/math.sqrt(sample_size))

In [17]:
t_test

-3.162277660168382

In [18]:
from scipy.stats import t

In [19]:
df = 40 - 1  

In [20]:
alpha = 0.05

In [21]:
t_critical = t.ppf(1 - alpha, df)

In [22]:
print("Critical value:", t_critical)

Critical value: 1.6848751217112248


<b>Conclusion</b>: The calculated t-value (−3.1623) is less than the critical value ( 1.68491). Therefore, we reject the null hypothesis. There is enough evidence to support the coffee shop's claim that their average wait time for customers is less than 5 minutes.

